data/lua/core/interface.lua

```diff
@@ -36,6 +37,50 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 		moving_unit:to_map(to_x, to_y)
 		wesnoth.wml_actions.redraw{}
 	end
+	---Fakes the move of a unit satisfying the given filter to position x, y.
+    ---Usable only during WML actions.
+	---Todo : maybe implement with C++.
+	---@param to_x integer
+    ---@param to_y integer
+    ---@param u unit
+	---@param queued boolean
+    function wesnoth.interface.move_unit_fake_queue(u, to_x, to_y, queued)
+        ---Fakes the move of a given unit, can be queued.
+		if to_x and to_y and u then
```

Comment:
Use `wesnoth.map.read_location` to get the actual x and y values from the given parameters. Something like this:

```lua
function f(unit, loc1, loc2)
local n, loc = wesnoth.map.read_location(loc1, loc2)
if u and loc then
```

Reply:
Never mind. I noticed that the way this is only called from C++ makes the use of `read_location` redundant here – in essence, it has already been handled in the C++ layer.